File: master-thesis-config-2/mtc-2-nodejs/algorithms/road-plan/logic/coordinate.js
class Coordinate {
  latitude;
  longitude;
  address;

  /**
   *  Coordinate Constructor
   */
  constructor(latitude, longitude, address) {
    this.latitude = latitude;
    this.longitude = longitude;
    this.address = address;
  }

  distance = (coordinate) => {
    return Math.sqrt(
      Math.pow(coordinate.latitude - this.latitude, 2) +
        Math.pow(coordinate.longitude - this.longitude, 2)
    );
  };
}

module.exports = Coordinate;

File: master-thesis-config-4/mtc-4-dotnet-static-cache/Domain/Algorithms/Coordinate.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Models;

namespace Domain.Algorithms
{
    [NotMapped]
    public class Coordinate
    {
        public int Order { get; set; } = 0;
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string Address { get; set; }

        public Coordinate(double latitude, double longitude, string address)
        {
            Latitude = latitude;
            Longitude = longitude;
            Address = address;
        }

        public double distance(Coordinate coordinate) =>
            Math.Sqrt(Math.Pow(coordinate.Latitude - Latitude, 2) + Math.Pow(coordinate.Longitude - Longitude, 2));

        public override string ToString()
        {
            return $"({Latitude}, {Longitude})";
        }
    }
}